src/connection.rs

```diff
@@ -286,3 +291,142 @@ where
         }
     }
 }
+
+pub struct Select<'a> {
+    cursor: Option<Cursor<'a>>,
+    error: Option<Error>,
+    columns_map: HashMap<String, usize>,
+}
+
+impl<'a> Select<'a> {
+    pub fn query(conn: &'a Connection, query: impl AsRef<str>) -> Self {
+        match conn.prepare(query) {
+            Ok(statement) => Self {
+                error: None,
+                columns_map: Self::create_columns_map(&statement),
+                cursor: Some(statement.into_cursor()),
+            },
+            Err(err) => Self {
+                cursor: None,
+                error: Some(err),
+                columns_map: Default::default(),
+            },
+        }
+    }
+
+    fn create_columns_map(statement: &Statement) -> HashMap<String, usize> {
+        (0..statement.column_count())
+            .map(|i| (statement.column_name(i).to_string(), i))
+            .collect()
+    }
+}
+
+impl<'a> From<Statement<'a>> for Select<'a> {
+    fn from(statement: Statement<'a>) -> Self {
+        Self {
+            error: None,
+            columns_map: Self::create_columns_map(&statement),
+            cursor: Some(statement.into_cursor()),
+        }
+    }
+}
+
+impl<'a> Iterator for Select<'a> {
+    type Item = Result<Row>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let columns_map = self.columns_map.clone();
+        match (self.cursor.as_mut(), self.error.take()) {
+            (_, Some(err)) => Some(Err(err)),
+            (Some(cursor), _) => match cursor.next() {
+                Err(err) => Some(Err(err)),
+                Ok(maybe_row) => Ok(maybe_row.map(|row| Row {
+                    row: row.to_vec(),
+                    columns_map,
+                }))
+                .transpose(),
+            },
+            (None, None) => None,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Row {
+    row: Vec<Value>,
```

Comment:
`row` in `Row`  does not sound good. `values` could be a better choice.

Reply:
Fixed in 9ff49e526630f0a5b34a4f689d90d7795f5d3f24